app/controllers/applications_controller.rb

```diff
@@ -20,10 +20,17 @@ def create
   def show
     @app = Application.find(id)
     if params[:q]
-      @search_results = Pet.where("name ILIKE ?", "%#{params[:q]}%").pluck(:name)
+      @search_results = Pet.where("name ILIKE ?", "%#{params[:q]}%")
     end
   end
 
+  def update
+    # use silent create here so if unique combination isn't entered, then just reloads page
+    PetApplication.create(application_id: params[:id], pet_id: params[:adopt])
+
```

Comment:
This is neat. As before for update we'd need to do a .find, .update. then .save - cleaner and more dry utilizing a silent create